core/services/dependencies/DependencyResolver.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace EventEspresso\core\services\dependencies;
+
+use EE_Dependency_Map;
+use EventEspresso\core\exceptions\InvalidAliasException;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\services\container\Mirror;
+use EventEspresso\core\services\loaders\ClassInterfaceCache;
+use ReflectionException;
+
+/**
+ * Class DependencyResolver
+ * Multiple classes that share the same common dependencies can have their dependencies
+ * automatically resolved using a DependencyResolver, and do not need to be registered by calling
+ * EE_Dependency_Map::registerDependencies() or manually adding an entry in the dependency map.
+ * Classes with more complex or unique dependencies should register them normally.
+ *
+ * @package EventEspresso\core\services\container
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+abstract class DependencyResolver implements DependencyResolverInterface
+{
+
+    /**
+     * @var Mirror $mirror
+     */
+    private $mirror;
+
+    /**
+     * @var ClassInterfaceCache $class_cache
+     */
+    private $class_cache;
+
+    /**
+     * @var EE_Dependency_Map $dependency_map
+     */
+    private $dependency_map;
+
+    /**
+     * @var ClassAlias[] $aliases
+     */
+    protected $aliases = array();
```

Comment:
Totally disagree since it is exactly what is intended. I think you may be misinterpreting the purpose of what this is for. 
I'll look at the phpdoc descriptions and see if I can clarify things, but I see no reason to rename it since this property is fully intended to be a list of aliases.